### ISSUE TITLE
Make the bundler treat project modules as npm modules

### ISSUE THREAD

**Issue by izaera** [liferay/liferay-js-toolkit#324]
Currently the transformation tasks for project files differ from those of external package modules. For example:

* `index.js` are not taken as default module files of folders
* you cannot put a package.json in a folder to designate the default module file
* ...

It would be great to make the bundler behave equally for coherence.

Of course, this may need some development in the server as well.

**Comment by ibhlool7** [comment]
Hi @izaera  
I've explored the source and found two packages where I can implement the logic (resolving index.js as default) 

`packages\babel-plugin-normalize-requires`

and 

`packages\babel-plugin-wrap-modules-amd`

where do you think, is appropriate for this?

**Comment by izaera** [comment]
@ibhlool7 The thing is that you cannot fix it here. It's the portal who needs to handle it. That's because:

1. In bundler 2, the imports feature: in case you are importing a module from some other OSGi bundle you don't know its structure.
2. In bundler 1, package changes: you may deploy a compatible new version of a package where index.js has been changed to other thing, so you need to resolve that during runtime (or at least deploy time). During build time you don't have enough information to do it.

**Comment by ibhlool7** [comment]
@izaera what do you mean from, handle it on the portal?
do you mean I can fix it on `liferay-amd-loader`?

**Comment by izaera** [comment]
@ibhlool7 Nope. On the portal side, in Java, where all the module processing takes place :point_right: https://github.com/liferay/liferay-portal/tree/master/modules/apps/frontend-js/frontend-js-loader-modules-extender

**Comment by wincent** [comment]
Going to do a mass-transfer of issues (including this one):

- **From:** https://github.com/liferay/liferay-js-toolkit/issues
- **To:** https://github.com/liferay/liferay-frontend-projects/issues

Note that many of the issues that we move may be stale, but we can close those once we've had time to review them. In the meantime, doing a bulk move makes is cheap and it will allow us to mark the liferay-js-toolkit repo as archived (read-only), which is an important final step in the consolidation process in order to avoid confusion.
